plugins/azure/fix_plugin_azure/resource/storage.py

```diff
@@ -945,130 +943,149 @@ def sku_filter(sku: AzureStorageSku) -> bool:
                 clazz=AzureStorageSku,
             )
 
-    @classmethod
-    def collect_usage_metrics(
-        cls: Type[MicrosoftResource], builder: GraphBuilder, collected_resources: List[MicrosoftResourceType]
-    ) -> None:
-        accounts = {storage_acc.id: storage_acc for storage_acc in collected_resources}
+    def collect_usage_metrics(self, builder: GraphBuilder) -> List[AzureMetricQuery]:
         queries = []
         start = builder.metrics_start
         now = builder.created_at
         delta = builder.metrics_delta
-        # Minimum interval and time range for storage metrics is 1H
+        # # Minimum interval and time range for storage metrics is 1H
         if delta.total_seconds() < 3600:
             delta = timedelta(hours=1)
             start = now - delta
-        for account_id in accounts:
-            blob_instance_id = account_id + "/blobServices/default"
-            file_instance_id = account_id + "/fileServices/default"
-            table_instance_id = account_id + "/tableServices/default"
-            queue_instance_id = account_id + "/queueServices/default"
-            queries.append(
-                AzureMetricQuery.create(
-                    metric_name="UsedCapacity",
-                    metric_namespace="microsoft.storage/storageaccounts",
-                    instance_id=account_id,
-                    aggregation=("average",),
-                    ref_id=account_id,
-                    unit="Bytes",
-                )
+        account_id = self.id
+        blob_instance_id = account_id + "/blobServices/default"
+        file_instance_id = account_id + "/fileServices/default"
+        table_instance_id = account_id + "/tableServices/default"
+        queue_instance_id = account_id + "/queueServices/default"
+
+        queries.append(
+            AzureMetricQuery.create(
+                metric_name="UsedCapacity",
+                metric_namespace="microsoft.storage/storageaccounts",
+                metric_normalization_name=MetricName.UsedCapacity,
+                instance_id=account_id,
+                aggregation=("average",),
+                ref_id=account_id,
+                unit="Bytes",
+                normalization=NormalizerFactory.bytes,
+                period=delta,
+                custom_start_time=start,
```

Comment:
Can we make this optional? and if not defined use the values from GraphBuilder? 
Same comment on all places where `period` and `custom_start_time` is taken directly from graphbuilder.